src/plugins/intel_cpu/src/nodes/kernels/registers_pool.hpp

```diff
@@ -300,49 +298,50 @@ class IsaRegistersPool<x64::avx512_core> : public RegistersPool {
+    IsaRegistersPool(std::initializer_list<Xbyak::Reg> regsToExclude) : IsaRegistersPool<dnnl::impl::cpu::x64::avx512_core>(regsToExclude) {}
+    IsaRegistersPool() : IsaRegistersPool<dnnl::impl::cpu::x64::avx512_core>() {}
 };
 
-template <x64::cpu_isa_t isa>
+template <dnnl::impl::cpu::x64::cpu_isa_t isa>
 RegistersPool::Ptr RegistersPool::create(std::initializer_list<Xbyak::Reg> regsToExclude) {
     return std::make_shared<IsaRegistersPool<isa>>(regsToExclude);
 }
 
 inline
-RegistersPool::Ptr RegistersPool::create(x64::cpu_isa_t isa, std::initializer_list<Xbyak::Reg> regsToExclude) {
+RegistersPool::Ptr RegistersPool::create(dnnl::impl::cpu::x64::cpu_isa_t isa, std::initializer_list<Xbyak::Reg> regsToExclude) {
 #define ISA_SWITCH_CASE(isa) case isa: return std::make_shared<IsaRegistersPool<isa>>(regsToExclude);
     switch (isa) {
-        ISA_SWITCH_CASE(x64::sse41)
-        ISA_SWITCH_CASE(x64::avx)
-        ISA_SWITCH_CASE(x64::avx2)
-        ISA_SWITCH_CASE(x64::avx2_vnni)
-        ISA_SWITCH_CASE(x64::avx512_core)
-        ISA_SWITCH_CASE(x64::avx512_core_vnni)
-        ISA_SWITCH_CASE(x64::avx512_core_bf16)
-        ISA_SWITCH_CASE(x64::avx512_core_fp16)
-        case x64::avx_vnni: return std::make_shared<IsaRegistersPool<x64::avx>>(regsToExclude);
-        case x64::avx512_core_bf16_ymm: return std::make_shared<IsaRegistersPool<x64::avx512_core>>(regsToExclude);
-        case x64::avx512_core_bf16_amx_int8: return std::make_shared<IsaRegistersPool<x64::avx512_core>>(regsToExclude);
-        case x64::avx512_core_bf16_amx_bf16: return std::make_shared<IsaRegistersPool<x64::avx512_core>>(regsToExclude);
-        case x64::avx512_core_amx: return std::make_shared<IsaRegistersPool<x64::avx512_core>>(regsToExclude);
-        case x64::avx512_vpopcnt: return std::make_shared<IsaRegistersPool<x64::avx512_core>>(regsToExclude);
-        case x64::isa_any:
-        case x64::amx_tile:
-        case x64::amx_int8:
-        case x64::amx_bf16:
-        case x64::isa_all:
+        ISA_SWITCH_CASE(dnnl::impl::cpu::x64::sse41)
+        ISA_SWITCH_CASE(dnnl::impl::cpu::x64::avx)
+        ISA_SWITCH_CASE(dnnl::impl::cpu::x64::avx2)
+        ISA_SWITCH_CASE(dnnl::impl::cpu::x64::avx2_vnni)
+        ISA_SWITCH_CASE(dnnl::impl::cpu::x64::avx512_core)
+        ISA_SWITCH_CASE(dnnl::impl::cpu::x64::avx512_core_vnni)
+        ISA_SWITCH_CASE(dnnl::impl::cpu::x64::avx512_core_bf16)
+        ISA_SWITCH_CASE(dnnl::impl::cpu::x64::avx512_core_fp16)
+        case dnnl::impl::cpu::x64::avx512_core_bf16_ymm:
+            return std::make_shared<IsaRegistersPool<dnnl::impl::cpu::x64::avx512_core>>(regsToExclude);
+        case dnnl::impl::cpu::x64::avx512_core_amx: return std::make_shared<IsaRegistersPool<dnnl::impl::cpu::x64::avx512_core>>(regsToExclude);
+        case dnnl::impl::cpu::x64::avx512_vpopcnt: return std::make_shared<IsaRegistersPool<dnnl::impl::cpu::x64::avx512_core>>(regsToExclude);
+        case dnnl::impl::cpu::x64::isa_undef:
```

Comment:
maybe just `default:` instead of explicit list below?

Reply:
Compiler is not happy when we forget about enum entries in scope of switch-case